pandas/_libs/tslibs/_timestamp.pyx

```diff
@@ -0,0 +1,420 @@
+        if PyDateTime_Check(result):
+            result = self.__class__(result)
+            result.nanosecond = self.nanosecond
+        return result
+
+    def __sub__(self, other):
+        if (is_timedelta64_object(other) or is_integer_object(other) or
+                PyDelta_Check(other) or hasattr(other, 'delta')):
+            # `delta` attribute is for offsets.Tick or offsets.Week obj
+            neg_other = -other
+            return self + neg_other
+
+        typ = getattr(other, '_typ', None)
+
+        # a Timestamp-DatetimeIndex -> yields a negative TimedeltaIndex
+        if typ in ('datetimeindex', 'datetimearray'):
+            # timezone comparison is performed in DatetimeIndex._sub_datelike
+            return -other.__sub__(self)
+
+        # a Timestamp-TimedeltaIndex -> yields a negative TimedeltaIndex
+        elif typ in ('timedeltaindex', 'timedeltaarray'):
+            return (-other).__add__(self)
+
+        elif other is NaT:
+            return NaT
+
+        # coerce if necessary if we are a Timestamp-like
+        if (PyDateTime_Check(self)
+                and (PyDateTime_Check(other) or is_datetime64_object(other))):
+            if isinstance(self, _Timestamp):
+                other = self.__class__(other)
+            else:
+                self = other.__class__(self)
+
+            # validate tz's
+            if not tz_compare(self.tzinfo, other.tzinfo):
+                raise TypeError("Timestamp subtraction must have the "
+                                "same timezones or no timezones")
+
+            # scalar Timestamp/datetime - Timestamp/datetime -> yields a
+            # Timedelta
+            try:
+                return self._create_timedelta(self.value - other.value)
+            except (OverflowError, OutOfBoundsDatetime):
+                pass
+
+        # scalar Timestamp/datetime - Timedelta -> yields a Timestamp (with
+        # same timezone if specified)
+        return datetime.__sub__(self, other)
+
+    cdef int64_t _maybe_convert_value_to_local(self):
+        """Convert UTC i8 value to local i8 value if tz exists"""
+        cdef:
+            int64_t val
+            npy_datetimestruct dts
+            int64_t delta
+            datetime dt
+
+        val = self.value
+        if self.tz is not None and not is_utc(self.tz):
+            # logic copied from _tz_convert_tzlocal_utc
+            # to prevent importing tslibs.conversion
+            dt64_to_dtstruct(val, &dts)
+            dt = datetime(dts.year, dts.month, dts.day, dts.hour,
+                          dts.min, dts.sec, dts.us)
+            # get_utcoffset (tz.utcoffset under the hood) only makes
+            # sense if dt is _wall time_, so convert to wall time
+            dt = dt.replace(tzinfo=tzutc())
+            dt = dt.astimezone(self.tz)
+            delta = (int(get_utcoffset(self.tz, dt).total_seconds())
+                     * 1000000000)
+            val = self.value + delta
```

Comment:
above 10 lines' logic (and a few cdefs above) copied from `_tz_convert_tzlocal_utc`, it was using `tz_convert_single` but with the args it was using it would always hand off the real effort to `_tz_convert_tzlocal_utc`.  The code copied here as also been made more efficient for the single case it needs to handle here.

Reply:
The "more efficient" looks like it is coming from not having the `if not to_utc` checks.  Is there anything else I'm missing?

Reply:
Nope, it's not really a big difference.  Saves a function and return call as well I guess? Pretty small